fix issues with multiple placeholders, generation of invalid URLs

JiriBalcar opened the pull request:
I this plugin was used with multiple placeholders like `/test/[slug1]/[slug2]` it appended base URL twice.
Also if some of the placeholders are not available, it replaced with 'unknown' which generated invalid URLs, I think it is better if those URLs are skipped.

Added support for strapi not in root directory.

Also added Vitest to test this behavior.

If you are OK with those changes, could I ask for build of new version of the plugin?

DominikProbst reviewed (changes requested):
Your change of the STRAPI_ADMIN_BACKEND_URL is a really good idea, but if the variable is not set in the .env-file, it will break the functionality. It would be better if you create a const in /admin/src/utils/helpers called BASE_URL or something like that that checks, if the variable is set and if not nothing ('') is used

like that:
`export const BASE_URL = (): string => process.env.STRAPI_ADMIN_BACKEND_URL && process.env.STRAPI_ADMIN_BACKEND_URL !== '/' ? process.env.STRAPI_ADMIN_BACKEND_URL : '';
`

In my test I always got "/" back, that's why you should check that too


And you made a lot of changes to the code that are only formatting stuff. not sure if that was intended from you

Thank you so far for your PR :-)

JiriBalcar commented:
Thank you @DominikProbst  for the review. I added the BASE_URL as you requested (nice catch). Also I reverted the formatting changes (autoformat on save) but maybe in the future it could be nice to have some formatting cleanup because now it is not that consistent (eg. " vs ')

DominikProbst commented:
Thank you. I merged it :-)

And yes, you're right about " and '. I will probably release a new version today in the evening that will fix that too
